Automattic/wp-calypso
# State: add getUnconnectedSiteUser selector and include it in the JPO steps.

This PR introduces an extra selector for the Jetpack Onboarding steps:
`getUnconnectedSiteUser()` retrieves the userEmail field of an onboarding site credentials by its ID and returns a hashed version of it.

This PR also incorporates the selector in the JPO steps.

Part of #21686.

**To test:**
- Checkout this branch
- Verify that the tests pass:

`npm run test-client client/state/selectors/test/get-unconnected-site-user-hash.js`

- also, follow the steps on https://github.com/Automattic/wp-calypso/pull/21626 to verify that the event tracking object contains all the properties set there.
